pkg/controller/installation/validation.go

```diff
@@ -247,6 +247,16 @@ func validateCustomResource(instance *operatorv1.Installation) error {
 			}
 		}
 
+		// VPP specific validation
+		if instance.Spec.CalicoNetwork.LinuxDataplane != nil && *instance.Spec.CalicoNetwork.LinuxDataplane == operatorv1.LinuxDataplaneVPP {
```

Comment:
We should probably also verify that the `spec.Variant` field is set to `Calico` since we don't support this for Calico Enterprise just yet. 

@tmjd any other validations you think we should be checking here?

Reply:
Are there any IPv4 and IPv6 limitations?
Any encapsulation limitations?
Are there any HostPorts or ContainerIPForwarding limitations?
I don't really know what limitations there would be, I was just looking at the fields in the Installation resource.

Reply:
Thanks, good questions :) 
IPv4 / IPv6 / dual stack is supported.
In terms of encap, we support flat IP / IPIP / VXLAN and wireguard so I think it should be alright.
HostPorts have been added recently, and ContainerIPForwarding should be supported but I don't think we've tested it, I'll make a note to check it.

Reply:
> HostPorts have been added recently,

Perhaps a nitty point, but does VPP support _disabling_ hostports? e.g., if `hostPorts: Disabled` is set, will VPP respect that? If not, we need to validate that hostPorts is never set to Disabled while in VPP mode.

Reply:
@AloysAugustin just one more question above, otherwise I think this is good to go

Reply:
We don't support disabling `HostPorts`, so I added a validation check for that.